### PR TITLE
use symfony process to run shell command

### PR DESCRIPTION
**Pull request by piotrkwiecinski**
Todo:
- Clean up imports
- Add dependency to composer.json
- Update exception docblock or map to LocalizedException

### CODE REVIEW

File: Css/PreProcessor/Adapter/Less/Processor.php [hunk 1]
```diff
@@ -93,29 +94,37 @@ public function processContent(AssetFile $asset)
...
+            $this->getPathToNodeBinary(),
+            ...$this->getNodeArgsAsArray(),
+            $this->getPathToLessCompiler(),
+            ...$this->getCompilerArgsAsArray(),
+            $filePath,
+        ]);
+
+        $process->run();
+
+        if (!$process->isSuccessful()) {
```
Comment (piotrkwiecinski) [inline Css/PreProcessor/Adapter/Less/Processor.php:107]: This bit is taken from the Process exception. I didn't want to expose a Symfony exception here as it's an implementation detail.

**Comment by hostep** [comment]
Thanks, I'm currently cleaning it up a bit so it works on older versions of PHP and Magento as to not break backwards compatibility and will then test it out a bit.
I'm also considering adding a configuration flag to allow or not allow errors, in case somebody updates this module and suddenly needs to rework their less code, that they have the ability to ignore errors and not crash their deploy flow.

I'm working on all this, on the following branch: https://github.com/baldwin-agency/magento2-module-less-js-compiler/commits/symfony-process/

**Comment by piotrkwiecinski** [comment]
It's good point about having a config option to display errors. I was considering it too. I just wanted to push the code to get your early feedback.

I forgot people still may use stores on PHP pre 8.

**Comment by hostep** [comment]
Alrighty, I think I have something that works, see the changes here: https://github.com/baldwin-agency/magento2-module-less-js-compiler/compare/v1-develop...symfony-process

Could you give it a try, by using the `dev-symfony-process` composer constraint?

It improves some things:
- stops compilation when something is wrong in the lessc process itself and outputs a bunch of info to the console
- introduces a `throw_on_error` config flag:
  - when disabled (default): outputs any warnings/errors lessc runs into while compiling to `var/log/system.log` (so no hard errors, but more like linting errors, the ones you ran into in #11)
  - when enabled: outputs those warnings/errors to the console when running SCD process or you'll see those same errors being outputted to css files when just browsing the website in developer mode and will see a more or less unstyled website
- also throws errors when less or css files are empty, see #8 

I've tested it with:
- Magento 2.1.16 | PHP 7.1 | symfony/process 2.8.52
- Magento 2.4.6-p6 | PHP 8.2 | symfony/process 5.4.23
- Magento 2.4.7-p1 | PHP 8.3 | symfony/process 6.4.8

**Comment by piotrkwiecinski** [comment]
@hostep I had a chance to test your changes with Magento 2.4.6-p6 | PHP 8.1 | symfony/process 5.4.23 and configuration works.

If you want a bit more flexibility for the flag you could use:
`filter_var($throwOnError, FILTER_VALIDATE_BOOLEAN)` in https://github.com/baldwin-agency/magento2-module-less-js-compiler/compare/v1-develop...symfony-process#diff-d25d3a8c2c8ef97d9372180d5c469a164e76d929607e073f11961bc847522763R303

Rest looks fine to me.

**Comment by hostep** [comment]
Thanks for the suggestion, that's now done: https://github.com/baldwin-agency/magento2-module-less-js-compiler/commit/6fe993eeaffa00b3754652cd51f8fcaa053cf7a7

This PR and some additional commits just got released in [version 1.7.0](https://github.com/baldwin-agency/magento2-module-less-js-compiler/releases/tag/v1.7.0), thank you very much for the feedback and testing here!

**Comment by piotrkwiecinski** [comment]
Awesome. Thank you for taking time to implement these changes.

**Pull request closed without merging**